src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASE_SIZE = 10;
+    private static final String INDEX_EXCEPTION_MESSAGE = "Provided index "
```

Comment:
Just use IndexOutOfBounds exception in method

Reply:
we have class by condition, you have same message in this exception throw
if (index < 0 || index > upperBound) {
            throw new ArrayListIndexOutOfBoundsException(
                    String.format("Index %d out of bounds for size %d", index, currentSize));
                    
                    use this instead

Reply:
> we have class by condition, you have same message in this exception throw if (index < 0 || index > upperBound) { throw new ArrayListIndexOutOfBoundsException( String.format("Index %d out of bounds for size %d", index, currentSize));
> 
> ```
>                 use this instead 
> ```

I think my realization is much simpler and readable. And it also provides enough information on the exception case. Readability etc.)